sopel/modules/instagram.py

```diff
@@ -8,34 +8,25 @@
 """
 from __future__ import unicode_literals, absolute_import, print_function, division
 
-try:
-    from ujson import loads
-except ImportError:
-    from json import loads
-from sopel.module import rule
-from sopel.tools import SopelMemory
-from requests import get
 from datetime import datetime
-import re
 
-instagram_regex = r'.*(https?:\/\/(?:www\.){0,1}instagram\.com\/([a-zA-Z0-9_\.]{,30}\/)?p\/[a-zA-Z0-9_-]+)\s?.*'
-instagram_pattern = re.compile(instagram_regex)
+from requests import get
 
+from sopel import module
 
-def setup(bot):
-    if not bot.memory.contains('url_callbacks'):
-        bot.memory['url_callbacks'] = SopelMemory()
-    bot.memory['url_callbacks'][instagram_pattern] = instaparse
+try:
+    from ujson import loads
+except ImportError:
+    from json import loads
 
 
-def shutdown(bot):
-    del bot.memory['url_callbacks'][instagram_pattern]
+instagram_regex = r'.*(https?:\/\/(?:www\.){0,1}instagram\.com\/([a-zA-Z0-9_\.]{,30}\/)?p\/[a-zA-Z0-9_-]+)\s?.*'
 
 # TODO: Parse Instagram profile page
 
 
-@rule(instagram_regex)
-def instaparse(bot, trigger):
+@module.url(instagram_regex)
+def instaparse(bot, trigger, match=None):
     # Get the embedded JSON
     json = get_insta_json(trigger.group(1))
```

Comment:
Shouldn't this use `match` instead of `trigger`, now that this function is using the `url` decorator?

Reply:
Complicated, since it would require a change in the `instagram_regex` I'm not comfortable with at the moment, in order to catch the whole URL in one group. But that's doable, I guess. Not sure I want to risk it for now.

Reply:
Marking resolved; the change to `@url` has been postponed.